src/v/storage/disk_log_impl.cc

```diff
@@ -2534,12 +2534,6 @@ ss::future<> disk_log_impl::remove_segment_permanently(
     _probe->delete_segment(*s);
     // background close
     s->tombstone();
-    if (s->has_outstanding_locks()) {
-        vlog(
-          stlog.info,
-          "Segment has outstanding locks. Might take a while to close:{}",
-          s->reader().filename());
-    }
```

Comment:
>  as now the lock is cleared before the segment is closed.

But isn't this called in many contexts (e.g. truncation, compaction), not just when the segment is being closed?